bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/RepositoryImpl.java

```diff
@@ -0,0 +1,145 @@
+        return new NodeImpl(nodeId, balVersion, balHome, System.getProperty("os.name"),
+                System.getProperty("os.version"));
+    }
+
+    private Artifact createArtifact(ObjectValue service, ObjectValue listener) {
+        ArtifactImpl artifact = new ArtifactImpl(service.toString(), Artifact.ArtifactType.SERVICE);
+        List<ObjectValue> listeners = (List<ObjectValue>) artifact.getDetail("listeners");
+        if (listeners == null) {
+            listeners = new ArrayList<>();
+            artifact.addDetail("listeners", listeners);
+        }
+        listeners.add(listener);
+        BServiceType serviceType = (BServiceType) TypeUtils.getImpliedType(service.getOriginalType());
+        artifact.addDetail("attachPoint", serviceType.attachPoint);
+        artifact.addDetail("resources", serviceType.getResourceMethods());
+        artifact.addDetail("remotes", serviceType.getRemoteMethods());
+        return artifact;
+    }
+
+    public static void addServiceListener(BObject listener, BObject service, Object attachPoint) {
+        BServiceType serviceType = (BServiceType) service.getType();
+        serviceType.attachPoint = attachPoint;
+        serviceListenerMap.put((ObjectValue) service, (ObjectValue) listener);
+        listenerServiceMap.put((ObjectValue) listener, (ObjectValue) service);
+    }
+
+    public static void addBallerinaInformation(String balHome, String balVersion) {
+        RepositoryImpl.balHome = balHome;
+        RepositoryImpl.balVersion = balVersion;
+    }
+
+    private static String generateNodeId() {
+        return UUID.randomUUID().toString();
+    }
+
+    /**
+     * The implementation of Ballerina runtime artifacts.
+     */
+    private static class ArtifactImpl extends Artifact {
+        private final Map<String, Object> details;
+
+        public ArtifactImpl(String name, ArtifactType type) {
+            super(name, type);
+            this.details = new HashMap<>();
+        }
+
+        private void addDetail(String detailsKey, Object value) {
+            this.details.put(detailsKey, value);
+        }
+
+        @Override
+        public Object getDetail(String detailKey) {
+            return details.getOrDefault(detailKey, null);
+        }
+    }
+
+    /**
+     * The implementation of Ballerina runtime node.
+     */
+    private static class NodeImpl extends Node {
+        private final Map<String, Object> details;
+
+        public NodeImpl(String nodeId, String balVersion, String balHome, String osName, String osVersion) {
+            super(nodeId);
+            this.details = new HashMap<>();
+            this.details.put("balVersion", balVersion);
+            this.details.put("balHome", balHome);
+            this.details.put("osName", osName);
+            this.details.put("osVersion", osVersion);
+        }
+
+        @Override
+        public Object getDetail(String detailKey) {
+            return details.getOrDefault(detailKey, null);
```

Comment:
Can we consider adding integration test to cover this line? Or we can add a test in the `remote.management` module.

Reply:
I added a small unit test to check the API call. We can add detailed tests inside the `remote.management` module.